### PR TITLE
Feature/project restructure

### PR DESCRIPTION
**Pull request by VaclavK**

### CODE REVIEW

**Comment by dguerin** [comment]
@VaclavK when I started this I was too minds about consuming the 2.3.0-alpha or not hence the few todos in the code. Now I see that 2.3.0-beta was released just today  https://www.nuget.org/packages/McMaster.Extensions.CommandLineUtils/2.3.0-beta Might be worth consuming this before we publish this.

**Comment by VaclavK** [comment]
@dguerin agreed, I did not notice the beta release.. leave that with me

**Comment by VaclavK** [comment]
AB#3508

File: src/EShopWorld.Tools/Commands/AutoRest/RenderProjectFileInternalCommand.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+using EShopWorld.Tools.Base;
+using EShopWorld.Tools.Commands.AutoRest.Models;
+using Microsoft.AspNetCore.Mvc.Razor;
+using Microsoft.AspNetCore.Mvc.ViewFeatures;
+
+namespace EShopWorld.Tools.Commands.AutoRest
+{
+    public class RenderProjectFileInternalCommand : AbstractRazorCommand<ProjectFileViewModel>
+    {
+
```
Comment (dguerin) [inline src/EShopWorld.Tools/Commands/AutoRest/RenderProjectFileInternalCommand.cs]: Line space 🗡

File: src/EShopWorld.Tools/Commands/AutoRest/Views/ProjectFile.cshtml [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+
+<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFrameworks>@String.Join(';', Model.TFMs)</TargetFrameworks>
+        <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+        <PackageRequireLicenseAcceptance>false</PackageRequireLicenseAcceptance>
+        <PackageId>eShopWorld.@(Model.ProjectName).Client</PackageId>
+        <Version>@Model.Version</Version>
+        <Authors>eShopWorld</Authors>
```
Comment (dguerin) [inline src/EShopWorld.Tools/Commands/AutoRest/Views/ProjectFile.cshtml]: Do we want to put in use the language latest property too? Any other properties that might be needed? ie: target framework

File: src/EShopWorld.Tools/Commands/KeyVault/KeyValueAccess.cs [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+    {       
+        internal static async Task<IList<SecretItem>> GetAllSecrets(string tenantId, string appId, string appSecret, string keyVaultName, string typeTagName, string nameTagName, string appName)
+        {
+            //authenticate using the  application id and secrets
+            var authContext = new AuthenticationContext($"https://login.windows.net/{tenantId}"); 
+            var credential = new ClientCredential(appId, appSecret);
+            var token = await authContext.AcquireTokenAsync("https://vault.azure.net", credential);
+
+            //open the vault
+            var kv = new KeyVaultClient(new TokenCredentials(token.AccessToken), new HttpClientHandler());
```
Comment (dguerin) [inline src/EShopWorld.Tools/Commands/KeyVault/KeyValueAccess.cs]: an MSI alternative here might be something to consider here, just a switch on the method or config which allows it to come from token or MSI.

File: src/EShopWorld.Tools/Commands/KeyVault/Views/PocoProjectFile.cshtml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+﻿@model EShopWorld.Tools.Commands.KeyVault.Models.GeneratePocoProjectViewModel
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
+    <PackageRequireLicenseAcceptance>false</PackageRequireLicenseAcceptance>
+    <PackageId>eShopWorld.@(Model.AppName).ConfigurationSecrets</PackageId>
+    <Version>@Model.Version</Version>
+    <Authors>eShopWorld</Authors>
+    <Company>eShopWorld</Company>
```
Comment (dguerin) [inline src/EShopWorld.Tools/Commands/KeyVault/Views/PocoProjectFile.cshtml]: any additional properties we might need here? use newest language, target framework etc.

File: src/EShopWorld.Tools/Properties/launchSettings.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
```
Comment (dguerin) [inline src/EShopWorld.Tools/Properties/launchSettings.json]: probably want to remove this launchsettings file.

File: src/EShopWorld.Tools/Commands/KeyVault/KeyValueAccess.cs (hunk 3 above)
Comment (VaclavK) [reply src/EShopWorld.Tools/Commands/KeyVault/KeyValueAccess.cs]: I was thinking the same actually... there was no MSI when this was done

File: src/EShopWorld.Tools/Commands/KeyVault/Views/PocoProjectFile.cshtml (hunk 4 above)
Comment (VaclavK) [reply src/EShopWorld.Tools/Commands/KeyVault/Views/PocoProjectFile.cshtml]: target framework there.. this is not a live (as in people coding actively in it) project - it merely hosts few pocos - so newest language would not bring value me thinks

File: src/EShopWorld.Tools/Commands/KeyVault/KeyValueAccess.cs (hunk 3 above)
Comment (VaclavK) [reply src/EShopWorld.Tools/Commands/KeyVault/KeyValueAccess.cs]: tracked as a story now

File: src/EShopWorld.Tools/Commands/AutoRest/Views/ProjectFile.cshtml (hunk 2 above)
Comment (VaclavK) [reply src/EShopWorld.Tools/Commands/AutoRest/Views/ProjectFile.cshtml]: target frameworks are actually configurable.. otherwise same as the other comment

**Pull request merged**